src/platform/DefaultPlatformFunctions.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Interface encapsulating various utility functions whose
+ * implementations depend on the platform / JS runtime.
+ */
+class DefaultPlatformFunctions {
+  safeExec(
+    cmd: string,
+    cb: (error: unknown, stdout: string | null) => void
+  ): void {
+    throw new Error('safeExec not implemented.');
```

Comment:
When will this surface? Should we abstract our the entire getUname function? I think it's the only consumer of exec that we have.